WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift

```diff
@@ -62,6 +85,21 @@ extension WPAnalytics {
         WPAnalytics.trackString(event.value, withProperties: mergedProperties)
     }
 
+
+    /// This will call each registered tracker and fire the given event.
+    /// - Parameters:
+    ///   - event: a `String` that represents the event name
+    ///   - properties: a `Hash` that represents the properties
+    ///   - blog: a `Blog` asssociated with the event
+    static func track(_ event: WPAnalyticsEvent, properties: [AnyHashable: Any], blog: Blog) {
+        var mergedProperties: [AnyHashable: Any] = event.defaultProperties ?? [:]
+        mergedProperties.merge(properties) { (_, new) in new }
+
+        mergedProperties[WPAppAnalyticsKeyBlogID] = blog.dotComID
+
+        WPAnalytics.trackString(event.value, withProperties: mergedProperties)
```

Comment:
I'm just wondering if we can call do something like this:

```
properties[WPAppAnalyticsKeyBlogID] = blog.dotComID
WPAnalytics.track(event, properties: propertiesWithBlog)
```

Then we don't need to repeat the "merge properties" logic. Wdyt?

Reply:
Thanks for the comment @leandroalonso I made a change for that ;-)